### ISSUE TITLE
esp8266

### ISSUE THREAD

**Issue by shelkovkin** [vsch/CLionArduinoPlugin#3]
I'm not sure if this should work. But I tried to configure to work with esp8266. By changing the board.txt to the one that comes with "https://github.com/esp8266/Arduino". As a result, I was able to create a new project for the desired board (by selecting from the list when creating it). But in the build configuration window, the list of boards remained the same, and the project is loaded with an error. It would be great if it would be possible to work with esp8266.
![01](https://user-images.githubusercontent.com/45637437/49536100-6b2f1800-f8ce-11e8-90a6-a2748fa37aa0.png)
![02](https://user-images.githubusercontent.com/45637437/49536116-76824380-f8ce-11e8-961b-3d9c5a1906be.png)
![03](https://user-images.githubusercontent.com/45637437/49536126-7b46f780-f8ce-11e8-8647-fa18ea398a8e.png)
![05](https://user-images.githubusercontent.com/45637437/49536136-7eda7e80-f8ce-11e8-8c7a-3a5e8db12247.png)
![06](https://user-images.githubusercontent.com/45637437/49536143-83069c00-f8ce-11e8-9cd8-227c9db9d258.png)

**Comment by vsch** [comment]
@shelkovkin, I have not tried it so it probably does not work. The original intention for the customized boards.txt path to point it to a newer Arduino one. In this case it is a third party boards.txt file.

I will have to play with it and looks like the primitive parser in the plugin is not parsing the third party boards.txt correctly. I will try it out. There has to be a way to make it work with third party boards.

**Comment by vsch** [comment]
@shelkovkin, I will have to add third party specific boards project handling. As I got myself a few esp8266 arduinos I have ability to test the full cycle. The third party boards.txt has different settings options and these need to be handled with that in mind.

**Comment by Janrupf** [comment]
I also would like to have the ability to use esp8266 so I can use CLion for my projects.

**Comment by PNixx** [comment]
And me to

**Comment by SingingBush** [comment]
Any update on this? I'd like to deploy to a WeMos D1 Mini from CLion.

**Comment by Janrupf** [comment]
Well, for now I would say use platform io, works pretty well and doesn't require any plugins

**Comment by dwalker-uk** [comment]
Did anyone manage to get this working, i.e. ESP8266-based Arduino code in CLion?  I have the same issue as the originator of this thread, i.e. I can get the right boards.txt to appear when I create the project, but doesn't seem to be anywhere to update the list of boards that is used in the Build Settings.

**Comment by Janrupf** [comment]
Yeah, you can easily use platform io as I mentioned (it effectively is Arduino)

**Comment by KamushekORIGINAL** [comment]
Any updates?
